### ISSUE TITLE
[BUG] Inconsistency: Status codes of the error classes don't match the response through `onError`

### ISSUE THREAD

**Issue by bogeychan** [elysiajs/elysia#188]
#### NotFoundError

https://github.com/elysiajs/elysia/blob/9cf9f865b629dce051fc74af20b84e637fdc64ba/src/error.ts#L31-L33

```ts
it('custom 404', async () => {
	const app = new Elysia().onError(({ code }) => {
		if (code === 'NOT_FOUND') {
			return 'UwU'
		}
	})

	const response = await app.handle(req('/not/found'))

	expect(await response.text()).toBe('UwU')
	expect(response.status).toBe(404)
})
```

```diff
+ Expected: 404
- Received: 200
```

#### ParseError

https://github.com/elysiajs/elysia/blob/9cf9f865b629dce051fc74af20b84e637fdc64ba/src/error.ts#L40-L42

```ts
it('custom 400 - ParseError', async () => {
	const app = new Elysia()
		.onError(({ code }) => {
			if (code === 'PARSE') {
				return 'UwU'
			}
		})
		.onParse(() => {
			throw new ParseError()
		})
		.post('/parse', ({ body }) => body)

	const response = await app.handle(post('/parse', {}))

	expect(await response.text()).toBe('UwU')
	expect(response.status).toBe(400)
})
```

```diff
+ Expected: 400
- Received: 500
```

### InternalServerError and ValidationError

 InternalServerError and ValidationError work as expected when https://github.com/elysiajs/elysia/pull/187 is merged.

https://github.com/elysiajs/elysia/blob/9cf9f865b629dce051fc74af20b84e637fdc64ba/src/error.ts#L31-L33

https://github.com/elysiajs/elysia/blob/9cf9f865b629dce051fc74af20b84e637fdc64ba/src/error.ts#L22-L24

**Comment by bogeychan** [comment]
the issue has already been mentioned here in a different form and still persists: https://github.com/elysiajs/elysia/issues/88#issuecomment-1712391230

**Comment by SaltyAom** [comment]
Fix with eabe264, available on 0.7.27.

Feel free to reopen if the issue still persists.

**Comment by bogeychan** [comment]
Works!

but `ParseError` still returns `500` instead of `400`:

https://github.com/elysiajs/elysia/blob/f9587a78b357ddac0d2aa236469392fbfcf5a202/src/error.ts#L42-L44

```ts
import { Elysia, ParseError } from 'elysia';

{
  const app = new Elysia()
    .onError(({ code, error }) => {
      if (code === 'PARSE') {
        console.log(error.status); // 400
        return 'UwU';
      }
    })
    .onParse(() => {
      throw new ParseError();
    })
    .post('/', ({ body }) => body);

  const response = await app.handle(
    new Request('http://localhost/', {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify({})
    })
  );

  console.log(await response.text()); // UwU
  console.log(response.status); // 500
}
```
